e2e/mv_test.go

```diff
@@ -311,3 +311,28 @@ func TestMoveMultipleS3ObjectsToS3(t *testing.T) {
 		assert.Assert(t, ensureS3Object(s3client, bucket, "dst/"+filename, content))
 	}
 }
+
+// mv s3://bucket/single*.blob* s3://bucket/single/
+func TestMoveSourceAndDestinationMatchWildcardWithTenThousandAndOneS3ObjectsToS3(t *testing.T) {
```

Comment:
i can't see how this test asserts that new files that are uploaded after the ListObjects request are ignored byt the List call.

Reply:
hmm, normally, if this test is executed it will lead to infinite loop/repetition, so we get timeout error; but following the changes the test case terminates, meaning files uploaded after ListObjects call are ignored. Otherwise, even if one is not ignored it will cause infinite repetition.